### DISCUSSION TITLE
How to find angles in degree for cards in image using imagemagick

### DISCUSSION THREAD

**Discussion by atfsolace** [ImageMagick/ImageMagick#2468] [Help]
Hi
IM version using: Version: ImageMagick 7.0.10-24 Q16 x64 2020-07-18 http://www.imagemagick.org

I want to find the card picture corners angels in degree using image magick

I do not have an idea how to start with this. Please help.

Image1: all red highlighted corners are not 90 degrees, So how to detect the angle of this area?
![image](https://user-images.githubusercontent.com/70207338/91277952-27b2c680-e7a1-11ea-9048-f44014d0bd5d.png)

Image2: original image of image1
![processed3](https://user-images.githubusercontent.com/70207338/91278448-dce57e80-e7a1-11ea-8f64-fe406fac6cee.jpeg)

Thank you in advanced.

**Comment by fmw42** [comment]
Can you explain why you need the angles? If we understand your problem better, perhaps we can suggest something.

**Reply by atfsolace** [reply]
Hi
Thank you for your reply.

I need the angles of cards in the image above because I need to check the card corners in the picture is proper or not, If I get the angles then its confirm that angle is 90 degree or not to check the card in the image is proper square or rectangle .

Thank you.

**Reply by atfsolace** [reply]
> Can you explain why you need the angles? If we understand your problem better, perhaps we can suggest something.

Please help for this If you have any solution.

Thank you

**Reply by fmw42** [reply]
Sorry, I do not have any good solution other than using Hough line transform after Canny edge detection. But that only finds lines and not intersections and angles for the intersections. You would have to post process the edges to find intersections and then compute the angles between the lines.

**Comment by snibgo** [comment]
From Image1 with the red lines, you could isolate those lines, then use morphology to find the coordinates of the ends and the corners. Trigonometry can then calculate the angles.

But I suppose you created the red lines manually, and you could have used your painting program to get those coordinates directly.

To automatically work from the original Image2, you could isolate just the brown border and white interior, then find the edge between those two areas. Then crop to each corner. This gives the equivalent of the red lines, so morphology and trigonometry can be used.
